pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -284,32 +281,31 @@ func extractSingleDependencyOfType(dependencies []wiringplugin.WiredDependency,
 	return matchedDependency, nil
 }
 
-func extractKubeComputeDependency(dependencies []wiringplugin.WiredDependency) (*apps_v1.Deployment, error) {
+func extractKubeComputeDependencyDetails(dependencies []wiringplugin.WiredDependency) (smith_v1.ResourceName, map[string]string, error) {
 	// Require exactly one KubeCompute dependency
 	kubeComputeDependency, err := extractSingleDependencyOfType(dependencies, apik8scompute.ResourceType)
 	if err != nil {
-		return nil, err
-	}
-
-	// Extract the deployment created by the KubeCompute dependency
-	var deploymentResource *smith_v1.Resource
-	for x, res := range kubeComputeDependency.SmithResources {
-		// TODO: Better way of identifying the correct deployment
-		// Because this could break if KubeCompute ever e.g. does Blue/Green deployments
-		if res.Spec.Object.GetObjectKind().GroupVersionKind() == k8s.DeploymentGVK {
-			deploymentResource = &kubeComputeDependency.SmithResources[x]
-			break
-		}
+		return "", nil, err
 	}
 
-	if deploymentResource == nil {
-		return nil, errors.New("failed to locate Kubernetes Deployment from KubeCompute dependency")
+	shapes := kubeComputeDependency.Contract.FindAllShapes(knownshapes.LabelledShape)
+	if len(shapes) == 0 {
+		return "", nil, errors.New("failed to locate Kubernetes Deployment from KubeCompute dependency")
 	}
 
-	deploymentSpec, ok := deploymentResource.Spec.Object.(*apps_v1.Deployment)
-	if !ok {
-		return nil, errors.New("cannot cast Deployment to expected spec type")
+	// Find labels attached to deployment object
+	// TODO: Better way of identifying the correct deployment
+	// Because this could break if KubeCompute ever e.g. does Blue/Green deployments
+	deploymentResourceName := wiringutil.ResourceName(kubeComputeDependency.Name)
```

Comment:
This assumption is just that - an assumption. Resource name can change and should not be relied on.